File: backend/entity/Patient.go
package entity
import (
	"gorm.io/gorm"
	"time"
)
		

type Patient struct{
	gorm.Model
	CitizenID string
	Prefix string
	FirstName string
	LastName string
	NickName string
	Enthnicity string
	Nationality string
	CongenitaDisease string
	BloodType string
	Gender string
	BirthDay time.Time
	PhoneNumber string
	Age int
	DrugAllergy string
}

type ContactPerson struct{
	gorm.Model
	Relationship string
	ContactperPhone string

	PatientID uint
	Patient Patient
}

type Address struct{
	gorm.Model
	HouseNumber string
	Moo string
	Subdistrict string
	District string
	Provice string
	Postcod string

	PatientID uint
	Patient Patient
}

type InitialSymptomps struct{
	gorm.Model
	Symptomps string
	BloodPressure string
	Visit time.Time
	HeartRate string
	weight float64
	Height float64
	
	ServiceID uint
	Service Service

	PatientID uint
	Patient Patient
}

type HistoryPatien struct{
	gorm.Model
	
	PatientID uint
	Patient []Patient

	ServiceID uint
	Service []Service

}

File: frontend/src/interface/patient.ts
//Database/interface/patient.ts
export interface Patient {
  id: number;
  citizenID: string;
  prefix: string;
  firstname: string;
  lastname: string;
  nickname: string;
  congenita_disease: string;
  blood_type: string;
  gender: string;
  birth_day: string;
  phonenumber: string;
  age: number;
  drugallergy: string;

  contactperson?: ContactPerson;
  address?: Address;
  initialsymptomps?: InitialSymptomps[];
  histories?: HistoryPatien[];
}

export interface ContactPerson {
  id: number;
  relationship: string;
  contactpersonphone: string;
  patientID: number;

  patient?: Patient;
}

export interface Address {
  id: number;
  housenumber: string;
  moo: string;
  subdistrict: string;
  district: string;
  provice: string;
  postcod: string;
  patientID: number;

  patient?: Patient;
}

export interface InitialSymptomps {
  id: number;
  symptomps: string;
  bloodpressure: string;
  visit: string; // ISO string
  heartrate: string;
  weight: number;
  height: number;
  serviceID: number;
  patientID: number;

  service?: ServiceRef;
  patient?: Patient;
}

export interface ServiceRef {
  id: number;
  name: string;
  price: number;
}

export interface HistoryPatien {
  id: number;
  patientID: number;
  caseDataID: number; // อ้างอิงเคสจากอีกระบบ/อีกตาราง
  note?: string;

  patient?: Patient;
  caseData?: CaseRef; // แนบรายละเอียดเคสตอน GET (optional)
}

export interface CaseRef {
  id: number;
  code: string;
  title: string;
  diagnosis?: string;
  price?: number;
}
